waku/v2/node/wakunode2.go

```diff
@@ -253,8 +255,12 @@ func New(opts ...WakuNodeOption) (*WakuNode, error) {
 		w.log.Error("creating localnode", zap.Error(err))
 	}
 
+	metadata := metadata.NewWakuMetadata(w.opts.clusterID, w.localNode, w.log)
```

Comment:
Isn't metadata protocol optional based on clusterID?
If so, shouldn't we initialize protocol only if clusterID is specified.

Reply:
I'll add this logic on the Start() of WakuMetadata. That way wakunode2 does not have to contain logic related to the value of the clusterId